_source/_posts/2020-04-29-master-filestream-in-csharp-aspnetcore.md

```diff
@@ -0,0 +1,321 @@
+
+I'm going to be using Okta for my security concerns. This makes it easy for me to plug into any type of .NET web application and manage users, roles, and other authentication features. I used to do all that stuff by hand but, over the years, I've been dealing with a lot of clients in banking, government, and other industries that have pretty demanding requirements around security. Okta takes one more thing off my plate, so I can focus on building business value.
+
+## Creating Your .NET Core 3.1 Web Project
+
+Create a new project in Visual Studio and start a new **ASP.NET Core Web Application**.
+
+{% img blog/filestream-csharp/06-new-web.png alt:"New VS Project" width:"800" %}{: .center-image }
+
+Choose a name for your project, then select your web project type. For this example I'll be creating a Model-View-Controller project but the setup will be similar for other types of web projects.
+
+{% img blog/filestream-csharp/07-MVC.png alt:"Select MVC" width:"800" %}{: .center-image }
+
+Select **ASP.NET Core 3.1** as the run-time.
+
+Let's add the Okta.AspNetCore package. Right-click *Dependencies* on the project, then click *Manage NuGet Packages*.
+
+Search for **Okta.AspNetCore** and click *Install*.
+
+{% img blog/filestream-csharp/09-Add-Okta-Nuget.png alt:"Okta AspNetCore Nuget" width:"800" %}{: .center-image }
+
+In the **Startup.cs** file, add the following using statement to the top of the file.
+
+```csharp
+using Okta.AspNetCore;
+```
+
+In your *ConfigureServices* method, add the following:
+
+```csharp
+var oktaMvcOptions = new OktaMvcOptions()
+{
+    OktaDomain = Configuration["Okta:Domain"],
+    ClientId = Configuration["Okta:ClientId"],
+    ClientSecret = Configuration["Okta:ClientSecret"],
+    Scope = new List<string> { "openid", "profile", "email" },
+};
+
+services.AddAuthentication(options =>
+{
+    options.DefaultAuthenticateScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+    options.DefaultSignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+    options.DefaultChallengeScheme = OktaDefaults.MvcAuthenticationScheme;
+})
+.AddCookie()
+.AddOktaMvc(oktaMvcOptions);
+```
+
+Add authorization to the *Configure* method, like so:
+
+```csharp
+app.UseAuthentication();
+```
+
+Lastly, in your *appsettings.json* file, make sure you have the following - filling in the brackets with the details of your application settings in Okta.
+
+```json
+"Okta": {
+    "ClientId": "{ClientId}",
+    "ClientSecret": "{ClientSecret}",
+    "Domain": "{Domain}"
+},
+```
```

Comment:
This post seems to be missing a whole section on creating an app on Okta and getting the values for this file. Did you QA this post to make sure it works? I've found that when I don't QA posts, they usually have mistakes.